src/shared/reboot-util.c

```diff
@@ -42,6 +43,9 @@ int update_reboot_parameter_and_warn(const char *parameter, bool keep) {
                 return 0;
         }
 
+        if (!ascii_is_valid(parameter) || strlen(parameter) > NAME_MAX)
```

Comment:
can you move this to its own helper? i.e. reboot_parameter_is_valid()?

Reply:
and then please use this in logind's method_set_reboot_parameter() to refuse the param early, before bothering with PK

Reply:
also, we expose JobTimeoutRebootArgument and RebootArgument in unit files, which configure the reboot argument to use in emergency handling cases. That currently uses simply config_parse_unit_string_printf(). please adjust to also use reboot_parameter_is_valid() before accepting the string.